paddlenlp/transformers/ernie/modeling.py

```
@@ -644,14 +644,10 @@ def forward(self,
         if attention_mask is None:
             attention_mask = paddle.unsqueeze(
                 (input_ids == self.pad_token_id
-                 ).astype(self.pooler.dense.weight.dtype) * -1e4,
+                 ).astype(paddle.get_default_dtype()) * -1e4,
```

`paddle.get_default_dtype` 不见得是和 weight 的dtype一致的，HF中mask的astype也是使用的weight的dtype

---

在 PretrainedModel 类中添加了 dtype 方法，默认获取 weight 的 dtype